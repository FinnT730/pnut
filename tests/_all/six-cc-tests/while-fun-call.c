int emit_line(int line, int f) {
  char c;
  putchar(line+ 48);
  putchar(':');
  putchar(' ');
  while ((c = fgetc(f)) && c != -1 && c != '\n') {
    putchar(c);
  }
  if (c != -1) {
    putchar('\n');
  }
  return c;
}

int main() {
  int f1;
  int f2;
  char c1;
  char c2;
  int i = 0;
  f1 = fopen("tests/_all/six-cc-tests/fgetc.c", "r");
  f2 = fopen("tests/_all/six-cc-tests/while-fun-call.c", "r");
  while (1) {
    c1 = emit_line(i, f1);
    c2 = emit_line(i, f2);
    if (c1 == -1 || c2 == -1){
      break;
    }
    i += 1;
  }

  fclose(f1);
  fclose(f2);
  return 0;
}
