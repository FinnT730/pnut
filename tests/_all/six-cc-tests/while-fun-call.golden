0: int main() {
0: int emit_line(int line, int f) {
1:   int f;
1:   char c;
2:   char c;
2:   putchar(line+ 48);
3:   f = fopen("tests/_all/six-cc-tests/fgetc.c", 0);
3:   putchar(':');
4:   while ((c = fgetc(f)) != -1) {
4:   putchar(' ');
5:     putchar(c);
5:   while ((c = fgetc(f)) && c != -1 && c != '\n') {
6:   }
6:     putchar(c);
7:   fclose(f);
7:   }
8: }
8:   if (c != -1) {
9: 9:     putchar('\n');
